FirebaseTestProject/Views/PostRow.swift

```diff
@@ -8,25 +8,92 @@
+    @State private var systemImage: String = ""
+    let deletePostAction:((Post) -> Void)
+    let isFavoriteAction:((Post) -> Void)
     
+    init(post: Binding<Post>, deletePostAction: @escaping (Post) -> Void, isFavoriteAction: @escaping (Post) -> Void) {
+        self._post = post
+        self.deletePostAction = deletePostAction
+        self.isFavoriteAction = isFavoriteAction
+    }
     
-   var body: some View {
-         VStack {
-             Text(post.title)
-                 .font(.title)
-                 .padding(.bottom, 8)
-             Text(post.text)
-                 .font(.body)
-                 .padding(.bottom, 8)
-             Text(post.author)
-                 .padding(.bottom, 8)
-         }
-   }
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack {
+                Text(post.title)
+                    .font(.largeTitle)
+                    .padding(.leading)
+                Spacer()
+                let userid = UserDefaults.standard.value(forKey: "userid") != nil ? UUID(uuidString: UserDefaults.standard.value(forKey: "userid") as! String) : UUID(uuidString: "00854E9E-8468-421D-8AA2-605D8E6C61D9")
+                if userid == post.authorid {
+                    Button(action: {
+                        showAlert = true
+                    }, label: {
+                        Label("", systemImage: "trash")
+                    })
+                    .buttonStyle(PlainButtonStyle())
+                }
+                Button(action: {
+                    Task {
+                        post.isFavorite = !post.isFavorite
+                        post.isFavorite ?
+                            try await PostService.favorite(post) : try await PostService.delete(favorite: post)
+                        isFavoriteAction(post)
+                    }
+                }, label: {
+                    Label("", systemImage: post.isFavorite ? "heart.fill" : "heart")
+                })
+                .padding()
+                .buttonStyle(PlainButtonStyle())
+            }
+            .alert(isPresented: $showAlert) {
+                Alert(title: Text("Do you want to delete this post?"),
+                      primaryButton: .cancel(),
+                      secondaryButton: .destructive(Text("Delete")) {
+                    deletePostAction(post)
+                })
+            }
+            HStack {
+                Text(post.author)
+                    .padding()
+                    .foregroundColor(Color(uiColor: UIColor.systemBlue))
+                    .onTapGesture {
+                        isPresenting = true
+                    }.sheet(isPresented: $isPresenting, onDismiss: {
+                        isPresenting = false
+                    }) {
+                        PostsList(viewStyle: .singleAuthor(post.author))
+                    }
+                Spacer()
+                Text(DateFormatter.postFormat(date: post.timestamp))
```

Comment:
`DateFormatter.postFormat(date: post.timestamp)` fails to compile with the following error:
> Type 'DateFormatter' has no member 'postFormat'

I know you created a `DateFormatter` extension with the `postFormat(date:)` method. If you include it in this PR, you should be good to go.

Reply:
Resolved by [commit 2c0f32b25cfd7e677d0a4549e34d9dab611ce4ef](https://github.com/benstone1/InstaCademy/pull/13/commits/2c0f32b25cfd7e677d0a4549e34d9dab611ce4ef)